### PR TITLE
[4138] coupon field missing fix

### PR DESCRIPTION
**Pull request by Mohamed-halim-sw**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4138

**Problem:**
* coupon component was missing on the checkout page.

**In this PR:**
* Checkout.component.js
	* line 471 rendered cartCoupon component

* CartCoupon.style.scss
	* line 43 to 46 styled the form to be block when computer screen is less than 1026px

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1. Coupon field is displayed on shipping page too. But it should display only on billing page as in default magenta
Screenshot 2022-02-03 at 13 25 44
2. Coupon field is displayed on billing step if user applied coupon in the cart
**Steps:**
Open PDP
Add to cart
Open Cart
Apply Coupon
Go to shipping step
Fill form
Go to billing step

**Actual result**
Coupon field is displayed 

**Expected result:**
Text' Applied coupon code: 50OFF' and button 'Remove coupon' are displayed as in screenshot
Screenshot 2022-02-03 at 14 08 20

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Fixed
3. Coupon block is displayed few sec on success complete order page
https://drive.google.com/file/d/1flgJaX4S4d-UiCslm92Rs-54Pywqemyh/view?usp=sharing

**Comment by aleksandrakorolova** [comment]
3. Fixed
4. Mobile: coupon block is displayed below payments methods
Screenshot 2022-02-08 at 07 59 57

**Expected result:**
it's displayed in summary block as in desktop view

**Comment by aleksandrakorolova** [comment]
Read carefully comments, move coupon field IN summary block

**Actual result:**
Coupon field is displayed below summary block


 **Expected result:**
 Coupon field is displayed in the summary block

**Comment by aleksandrakorolova** [comment]
1. Coupon should be displayed in Summary block only in mobile view. On desktop view - coupon field is displayed below summary block
2. Coupon field is unwrapped in mobile. 
**Expected result:**
Coupon field is always wrapped in mobile

**Comment by aleksandrakorolova** [comment]
5. Fixed
6. Fixed
7. Mobile: Text 'Have a discount code?' is missing above coupon field
Screenshot 2022-02-11 at 08 00 01

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 1]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
...
+                  heading={ __('Have a discount code?') }
+                  mix={ { block: 'CartPage', elem: 'Discount' } }
+                  isArrow
+                >
+                    <CartCoupon couponCode={ coupon_code } />
+                </ExpandableContent>
+            );
+        }
+
+        return null;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:164]: Make it simplier, can even create separate function for rendering mobile version.

1. checkout step is not billing step => return null;
2. if is mobile => render mobile version;
3. default render just expandablecontent.

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 2]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
...
+
+        if (isMobile && checkoutStep === BILLING_STEP) {
+            return (
+                <div>
+                    <div
+                      block="ExpandableContent"
+                      elem="Heading"
+                      mix={ { block: 'CartPage', elem: 'ExpandableContentHeading' } }
+                    >
+                    Have a discount code?
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:145]: Translation?

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 3]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
...
+                    <CartCoupon couponCode={ coupon_code } />
+                </div>
+            );
+        }
+
+        if (checkoutStep === BILLING_STEP) {
+            return (
+                <ExpandableContent
+                  heading={ __('Have a discount code?') }
+                  mix={ { block: 'CartPage', elem: 'Discount' } }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:156]: Is block correct?

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 4]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
...
+            return null;
+        }
+
+        if (isMobile && checkoutStep === BILLING_STEP) {
+            return (
+                <div>
+                    <div
+                      block="ExpandableContent"
+                      elem="Heading"
+                      mix={ { block: 'CartPage', elem: 'ExpandableContentHeading' } }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:143]: Is block correct?

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 5]
```diff
@@ -327,6 +330,31 @@ export class Checkout extends PureComponent {
...
+        } = this.props;
+
+        if (!items || items.length < 1) {
+            return null;
+        }
+
+        if (checkoutStep === BILLING_STEP) {
+            return (
+                <ExpandableContent
+                  heading={ __('Have a discount code?') }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:346]: Is block correct?

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 6]
```diff
@@ -327,6 +330,31 @@ export class Checkout extends PureComponent {
...
+                  mix={ { block: 'CartPage', elem: 'Discount' } }
+                  isArrow
+                >
+                    <CartCoupon couponCode={ coupon_code } />
+                </ExpandableContent>
+            );
+        }
+
+        return null;
+    }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:356]: if checkout step not billing, return null before.

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 7]
```diff
@@ -358,15 +387,19 @@ export class Checkout extends PureComponent {
...
               checkoutStep={ checkoutStep }
               totals={ checkoutTotals }
               paymentTotals={ paymentTotals }
               isExpandable={ isMobile }
+              onCouponCodeUpdate={ onCouponCodeUpdate }
               // eslint-disable-next-line react/jsx-no-bind
               renderCmsBlock={ () => this.renderPromo(true) }
               showItems
             />
+            { !showOnMobile && this.renderDiscountCode() }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:401]: Indent out here is not correct.

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 8]
```diff
@@ -117,6 +123,51 @@ export class CheckoutOrderSummary extends PureComponent {
...
+
+        if (!items || items.length < 1) {
+            return null;
+        }
+
+        if (checkoutStep !== BILLING_STEP) {
+            return null;
+        }
+
+        if (isMobile && checkoutStep === BILLING_STEP) {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:156]: U actually know that it is billing step

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 9]
```diff
@@ -117,6 +123,51 @@ export class CheckoutOrderSummary extends PureComponent {
...
+            isMobile
+        } = this.props;
+
+        if (!items || items.length < 1) {
+            return null;
+        }
+
+        if (checkoutStep !== BILLING_STEP) {
+            return null;
+        }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:154]: unify

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 10]
```diff
@@ -326,6 +377,7 @@ export class CheckoutOrderSummary extends PureComponent {
     }
 
     renderExpandableContent() {
+        const { isMobile } = this.props;
         return (
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:381]: space before return

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 11]
```diff
@@ -334,6 +386,7 @@ export class CheckoutOrderSummary extends PureComponent {
                 { this.renderItems() }
                 { this.renderCmsBlock() }
                 { this.renderTotals() }
+                { isMobile && this.renderDiscountCode() }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:389]: Im confused. Out here u check if mobile and then render. Why in this case in renderDiscountCode function it again checks if mobile and renders depending on it.

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 12]
```diff
@@ -327,6 +330,31 @@ export class Checkout extends PureComponent {
...
+            checkoutStep
+        } = this.props;
+
+        if (!items || items.length < 1) {
+            return null;
+        }
+
+        if (checkoutStep !== BILLING_STEP) {
+            return null;
+        }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:345]: unify

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 13]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
         );
     }
 
+    renderMobileDiscount(coupon_code) {
+        return (
+            <div>
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:128]: use just fragment

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 14]
```diff
@@ -358,15 +383,19 @@ export class Checkout extends PureComponent {
...
-            />
+            <>
+                <CheckoutOrderSummary
+                  checkoutStep={ checkoutStep }
+                  totals={ checkoutTotals }
+                  paymentTotals={ paymentTotals }
+                  isExpandable={ isMobile }
+                  onCouponCodeUpdate={ onCouponCodeUpdate }
+                // eslint-disable-next-line react/jsx-no-bind
+                  renderCmsBlock={ () => this.renderPromo(true) }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:394]: Fix it please

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 15]
```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
...
 
+    renderMobileDiscount(coupon_code) {
+        return (
+            <>
+                <div
+                  block="ExpandableContent"
+                  elem="Heading"
+                  mix={ { block: 'CheckoutOrderSummary', elem: 'ExpandableContentHeading' } }
+                >
+                { __('Have a discount code?') }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:134]: Indent out here is not correct

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 16]
```diff
@@ -355,24 +379,30 @@ export class Checkout extends PureComponent {
...
-              showItems
-            />
+            <>
+                <CheckoutOrderSummary
+                  checkoutStep={ checkoutStep }
+                  totals={ checkoutTotals }
+                  paymentTotals={ paymentTotals }
+                  isExpandable={ isMobile }
+                  onCouponCodeUpdate={ onCouponCodeUpdate }
+                // eslint-disable-next-line react/jsx-no-bind
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:400]: remove eslint disable

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**